shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -236,28 +242,26 @@ - (void)sharedSetupWithProject:(nullable FlutterDartProject*)project
   // Need the project to get settings for the view. Initializing it here means
   // the Engine class won't initialize it later.
   if (!project) {
-    project = [[[FlutterDartProject alloc] init] autorelease];
+    project = [[FlutterDartProject alloc] init];
   }
   FlutterView.forceSoftwareRendering = project.settings.enable_software_rendering;
   _weakFactory = std::make_unique<fml::WeakNSObjectFactory<FlutterViewController>>(self);
-  auto engine = fml::scoped_nsobject<FlutterEngine>{[[FlutterEngine alloc]
-                initWithName:@"io.flutter"
-                     project:project
-      allowHeadlessExecution:self.engineAllowHeadlessExecution
-          restorationEnabled:[self restorationIdentifier] != nil]};
-
+  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"io.flutter"
+                                                      project:project
+                                       allowHeadlessExecution:self.engineAllowHeadlessExecution
+                                           restorationEnabled:self.restorationIdentifier != nil];
   if (!engine) {
     return;
   }
 
   _viewOpaque = YES;
   _engine = engine;
-  _flutterView.reset([[FlutterView alloc] initWithDelegate:_engine
-                                                    opaque:self.isViewOpaque
-                                           enableWideGamut:project.isWideGamutEnabled]);
-  [_engine.get() createShell:nil libraryURI:nil initialRoute:initialRoute];
+  _flutterView = [[FlutterView alloc] initWithDelegate:_engine
+                                                opaque:_viewOpaque
+                                       enableWideGamut:project.isWideGamutEnabled];
+  [_engine createShell:nil libraryURI:nil initialRoute:initialRoute];
   _engineNeedsLaunch = YES;
-  _ongoingTouches.reset([[NSMutableSet alloc] init]);
+  _ongoingTouches = [[NSMutableSet alloc] init];
   [self loadDefaultSplashScreenView];
   [self performCommonViewControllerInitialization];
```

Comment:
And these are effectively init code too. And `loadDefaultSplashScreenView` is calling `setSplashScreenView:`, so we're using a setter to initialize an ivar from `init*`. And `performCommonViewControllerInitialization` calls another `self` method that in turn calls three more `self` methods (although at least those should be easy to move to static functions, it looks like).

This is a real mess, and we should definitely have a TODO with a tracking issue.

Reply:
Done.